File: frontend/src/types/vocabulary.ts
import type { ApiResponse } from './quizHistory'

export type { ApiResponse }

export interface VocabularyEntry {
  id: number
  japanese: string
  kana: string
  accent: number | null
  chinese: string
  created_at: string
}

export interface VocabularyCreate {
  japanese: string
  kana: string
  accent: number | null
  chinese: string
}

export type VocabularyUpdate = VocabularyCreate

File: backend/app/schemas/vocabulary.py
from datetime import datetime

from pydantic import BaseModel


# --- Input schemas ---

class VocabularyCreate(BaseModel):
    japanese: str
    kana: str
    accent: int | None = None
    chinese: str


class VocabularyUpdate(BaseModel):
    japanese: str
    kana: str
    accent: int | None = None
    chinese: str


# --- Output schemas ---

class VocabularyResponse(BaseModel):
    id: int
    japanese: str
    kana: str
    accent: int | None
    chinese: str
    created_at: datetime

    model_config = {"from_attributes": True}
